### ISSUE TITLE
Prune openssl

### ISSUE THREAD

**Issue by Xanewok** [paritytech/sccache#87]
There was an attempt to remove openssl in favour of pure Rust implementation at https://github.com/paritytech/sccache/pull/67, however we still include it via `rouille` > `tiny_http` implementation. We track moving off `rouille` at https://github.com/paritytech/sccache/issues/39.

The benefit, as I understand it, is to make the build and distribution easier or more portable, however we do build and ship the binary by using the `openssl/vendored` feature already in [Actions](https://github.com/paritytech/sccache/blob/e5d16a16de5c9e996385c5ec014be710af103a07/.github/workflows/ci.yml#L251) and [GitLab](https://github.com/paritytech/sccache/blob/e5d16a16de5c9e996385c5ec014be710af103a07/.gitlab-ci.yml#L115), so I'm not sure how critical this issue is. 

@drahnr @TriplEight is it something that we want to resolve in the nearest future?

**Comment by drahnr** [comment]
I agree that this not an immediate priority, #67 will resolve this incidentally.

Generally I consider it an advantage to avoid openssl entirely, mostly due to the fact of requiring a C++ compiler and the general topic of reproducibility. It's a nice to have, eventually.

**Comment by TriplEight** [comment]
I'm in favour of that, I will free us from having `openssl` as a dependency in many places.

**Comment by Xanewok** [comment]
As I understand, after merging #131 we're still blocked upstream by https://github.com/briansmith/webpki/issues/54 (we can't use rustls without IP support); is that correct @montekki?
